src/core/algorithms/fd/sfd/frequency_handler.cpp

```diff
@@ -0,0 +1,49 @@
+#include "frequency_handler.h"
+
+#include <algorithm>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "model/table/column_index.h"
+#include "model/table/tuple_index.h"
+#include "model/table/typed_column_data.h"
+
+namespace algos {
+void FrequencyHandler::InitFrequencyHandler(std::vector<model::TypedColumnData> const &data,
+                                            model::ColumnIndex columns,
+                                            size_t max_amount_of_categories) {
+    cardinality_.resize(columns, 0);
+    frequency_maps_.resize(columns);
+    freq_sums_.resize(columns, 0);
+    for (model::ColumnIndex col_ind = 0; col_ind < data.size(); col_ind++) {
+        std::unordered_map<std::string, size_t> value_frequency_counter;
+        auto const &col_data = data[col_ind];
+        for (model::TupleIndex row_ind = 0; row_ind < col_data.GetNumRows(); row_ind++) {
+            auto &freq = value_frequency_counter[col_data.GetDataAsString(row_ind)];
```

Comment:
Was it neccessary to view values as strings? Can't we save some memory and time by only looking at their byte representation and storing pointers to already existing values?

Reply:
In order to use stl containers with byte representation properly we will have to do something like this  (at least that's the only thing I could think of.):
```cpp
#pragma once
#include <functional>

#include "model/types/types.h"

namespace algos {
struct TypedByte {
    std::byte const* data;
    model::Type const& type;
    TypedByte() = delete;
    TypedByte(std::byte const* data, model::Type const& type) : data(data), type(type) {}

    std::size_t GetHash() const {
        return this->type.GetHasher()(this->data);
    }

    TypedByte& operator=(TypedByte const& rhs) {
        if (this != &rhs) {
            data = rhs.data;
        }
        return *this;
    }

};
}  // namespace algos

template <>
struct std::hash<algos::TypedByte> {
    std::size_t operator()(algos::TypedByte const& tb) const noexcept {
        return tb.GetHash();
    }
};

template <>
struct std::equal_to<algos::TypedByte> {
    std::size_t operator()(algos::TypedByte const& lhs,
                           algos::TypedByte const& rhs) const noexcept {
        return lhs.type.Compare(lhs.data, rhs.data) == model::CompareResult::kEqual;
    }
};
```
and do something like this to all the used containers:
```cpp
std::unordered_map<TypedByte, size_t> value_frequency_counter;
```
It works fine with all the types introduced in Desbordante except for `EmptyType` which throws an error whenever hash or comparison function is called. So it seems to me that there is no way to avoid using strings here.

Reply:
Do we actually even need types?

Reply:
In order to implement hash and comparison functions on byte representation itself, without using typed implementations, we need to know the length of the byte sequence underneath `std::byte*`, which i don't know how to retrieve.

Reply:
I mostly talking about non-typed ColumnData, but ok